code/processes/wdb.q

```diff
@@ -102,149 +107,212 @@ mergemaxrows:{[tabname] mergenumrows^mergenumtab[tabname]}
 tablelist:{[] sortedlist:exec tablename from `bytes xdesc .wdb.tabsizes;
-	(sortedlist union tables[`.]) except ignorelist}
+    (sortedlist union tables[`.]) except ignorelist}
 
 /- function to upsert to specified directory
-upserttopartition:{[dir;tablename;tabdata;pt;expttype;expt]	    		
-	.lg.o[`save;"saving ",(string tablename)," data to partition ",
-		/- create directory location for selected partiton
-		string directory:` sv .Q.par[dir;pt;tablename],
-		/- replace random chracters in symbols with _
-		(`$"_"^.Q.an .Q.an?"_" sv string 
-		/- convert to symbols and replace any null values with `NONE
-		`NONE^ -1 _ `${@[x; where not ((type each x) in (10 -10h));string]} expt,(::)),`];	
-	/- upsert selected data matched on partition to specific directory 	
-	.[
-		upsert;
-		(directory;r:?[tabdata;{(x;y;(),z)}[in;;]'[expttype;expt];0b;()]);		
-		{[e] .lg.e[`savetablesbypart;"Failed to save table to disk : ",e];'e}
-	];
-	.lg.o[`track;"appending details to partsizes"];
-	/-key in partsizes are directory to partition, need to drop training slash in directory key
-	.merge.partsizes[first ` vs directory]+:(count r;-22!r);
-	};
-	
+upserttopartition:{[dir;tablename;tabdata;pt;expttype;expt]
+    .lg.o[`save;"saving ",(string tablename)," data to partition ",
+        /- create directory location for selected partiton
+        string directory:` sv .Q.par[dir;pt;tablename],
+        /- replace random chracters in symbols with _
+        (`$"_"^.Q.an .Q.an?"_" sv string
+        /- convert to symbols and replace any null values with `NONE
+        `NONE^ -1 _ `${@[x; where not ((type each x) in (10 -10h));string]} expt,(::)),`];
+    /- upsert selected data matched on partition to specific directory
+    .[
+        upsert;
+        (directory;r:?[tabdata;{(x;y;(),z)}[in;;]'[expttype;expt];0b;()]);
+        {[e] .lg.e[`savetablesbypart;"Failed to save table to disk : ",e];'e}
+    ];
+    .lg.o[`track;"appending details to partsizes"];
+    /-key in partsizes are directory to partition, need to drop training slash in directory key
+    .merge.partsizes[first ` vs directory]+:(count r;-22!r);
+    };
+
+/- function to upsert to specified directory using enumerated extra partitioning
+upserttopartitionenum:{[dir;tablename;tabdata;pt;expttype;expt]
+    hdbsym:` sv hdbsettings[`hdbdir],`sym;
+    /- enumerate current extra partition agains the hdb sym file
+    i:get[hdbsym]?hdbsym?first expt;
+    .lg.o[`save;"saving ",(string tablename)," data to partition ",
+                /- create directory location for selected partiton
+                string directory:` sv .Q.par[dir;pt;`$string i],tablename,`];
+    /- selecting rows of table with the corresponding symbol
+    r:?[tabdata;enlist(in;first expttype;expt);0b;()];
+    /- upsert selected data matched on partition to specific directory
+    .[
+     upsert;
+     (directory;r);
+     {[e] .lg.e[`savetablesbypartenum;"Failed to save table to disk : ",e];'e}
+     ];
```

Comment:
I think breaking this over so many lines makes it less readable (it's not immediately clear that these lines are a single expression). Perhaps:

```suggestion
    .[upsert;(directory;r);
      {[e] .lg.e[`savetablesbypartenum;"Failed to save table to disk : ",e];'e}];
```

Reply:
Yeah done!